tests/relay/test_node.py

```diff
@@ -0,0 +1,291 @@
+from typing import Any, List, Optional
+
+import pytest
+import strawberry
+from sqlalchemy import Column, Integer, String
+from sqlalchemy.engine import Engine
+from sqlalchemy.ext.asyncio.engine import AsyncEngine
+from sqlalchemy.orm import sessionmaker
+from strawberry import relay
+from strawberry_sqlalchemy_mapper import StrawberrySQLAlchemyMapper, node
+
+
+@pytest.fixture
+def fruit_table(base: Any):
+    class Fruit(base):
+        __tablename__ = "fruit"
+        id = Column(Integer, autoincrement=True, primary_key=True)
+        name = Column(String(50), nullable=False)
+        color = Column(String(50), nullable=False)
+
+    return Fruit
+
+
+def test_node(
+    base: Any,
+    engine: Engine,
+    sessionmaker: sessionmaker,
+    fruit_table,
+):
+    base.metadata.create_all(engine)
+    mapper = StrawberrySQLAlchemyMapper()
+
+    @mapper.type(fruit_table)
+    class Fruit(relay.Node):
+        id: relay.NodeID[int]
+
+    @strawberry.type
+    class Query:
+        fruit: Fruit = node(sessionmaker=sessionmaker)
+
+    schema = strawberry.Schema(query=Query)
+
+    query = """\
+    query Fruit($id: GlobalID!) {
+      fruit(id: $id) {
+        id
+        name
+        color
+      }
+    }
+    """
+
+    with sessionmaker() as session:
+        f1 = fruit_table(name="Banana", color="Yellow")
+        f2 = fruit_table(name="Apple", color="Red")
+        f3 = fruit_table(name="Orange", color="Orange")
+
+        session.add_all([f1, f2, f3])
+        session.commit()
+
+        for f in [f1, f2, f3]:
+            result = schema.execute_sync(query, {"id": relay.to_base64("Fruit", f.id)})
+            assert result.errors is None
+            assert result.data == {
+                "fruit": {
+                    "id": relay.to_base64("Fruit", f.id),
```

Comment:
Trying to understand how this relay API works. Having this returned in `id` seems like it would obfuscate `id` to the caller, which for a lot of situations is totally fine, but what if we wanted to do this without having an opaque `id`? Is there a way to put the relay ID in a different response field, but still tell the mapper that it should use `id` for mapping?

Reply:
Since this is the resolved query to the frontend, the frontend clients should only know the real relay ID. In your mutations you can use `relay.from_base64` and then the id to node function to get the corresponding DB entity. Where else would you need the raw ID where it is difficult to access by this paradigm?

Not sure I got the question right, so please feel free to specify.

Reply:
It's more a matter of, what if someone already has a field named ID that the client is already using, but they want to add support using this new relay functionality. It's nice to think all ids should be opaque, but that's far too opinionated for a middleware library like this to make that decision IMO.

So my question is, what does someone do if they already have a field named "id" that the client is depending on? Can they put this relay ID in a different field so as not to clobber "id"?

Reply:
@mattalbr it is part of the `Node` interface and its spec to have a `id` attribute which is serialized something that can globally identify the object (i.e. without even having to tell the type).

Having said that, there's a proposal to make it easier to support custom implementations (in case you want to make it opaque in some other way, or even skip it at all) which might be useful once it gets merged: https://github.com/strawberry-graphql/strawberry/issues/3057

Another possibility is to expose the original id in another field.

But I would say that anyone wanting to use this is not only aware of the whole GlobalID thing, but also expects that it works like that.

Reply:
> But I would say that anyone wanting to use this is not only aware of the whole GlobalID thing, but also expects that it works like that.

I personally disagree with this one :D I think the global id is mostly useful when using relay as a client, which might not be the case most of the times 😊

I think it might be worth trying to solve 3057 in Strawberry, but I haven't had the right amount of time to think about a good solution yet